src/main/java/uk/gov/hmcts/reform/logging/layout/ReformLoggingLayout.java

```diff
@@ -0,0 +1,94 @@
+package uk.gov.hmcts.reform.logging.layout;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.classic.spi.ThrowableProxy;
+import ch.qos.logback.core.CoreConstants;
+import ch.qos.logback.core.LayoutBase;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.hmcts.reform.logging.exception.AbstractLoggingException;
+import uk.gov.hmcts.reform.logging.exception.InvalidExceptionImplementation;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.NoSuchElementException;
+
+public class ReformLoggingLayout extends LayoutBase<ILoggingEvent> {
+
+    private final Logger log = LoggerFactory.getLogger(ReformLoggingLayout.class);
+
+    private DateTimeFormatter dateFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS");
+
+    /**
+     * By default require thread to be logged.
+     */
+    private boolean requireThread = true;
+
+    /**
+     * By default require an AlertLevel tag to be present for any exception log.
+     */
+    private boolean requireAlertLevel = true;
+
+    public void setDateFormat(String format) {
+        dateFormat = DateTimeFormatter.ofPattern(format);
+    }
+
+    public void setRequireThread(boolean requireThread) {
+        this.requireThread = requireThread;
+    }
+
+    public void setRequireAlertLevel(boolean requireAlertLevel) {
+        this.requireAlertLevel = requireAlertLevel;
+    }
+
+    private void triggerBadImplementationLog(Throwable cause) {
+        Throwable invalid = new InvalidExceptionImplementation("AlertLevel is mandatory as per configuration", cause);
+
+        log.error("Bad implementation of '" + cause.getClass().getCanonicalName() + "' in use", invalid);
+    }
+
+    @Override
+    public String doLayout(ILoggingEvent event) {
+        AbstractLoggingException exception = null;
+
+        if (requireAlertLevel && event.getLevel().isGreaterOrEqual(Level.ERROR)) {
+            Throwable eventException = ((ThrowableProxy) event.getThrowableProxy()).getThrowable();
+
+            if (eventException instanceof AbstractLoggingException) {
+                exception = (AbstractLoggingException) eventException;
+            } else  {
+                triggerBadImplementationLog(eventException);
+            }
+        }
+
+        Instant instant = Instant.ofEpochMilli(event.getTimeStamp());
+        LocalDateTime dateTime = LocalDateTime.ofInstant(instant, ZoneId.systemDefault());
+        String log = dateFormat.format(dateTime);
+
+        log += " " + String.format("%-5s", event.getLevel().levelStr);
```

Comment:
use `StringBuilder` instead of all this concatenation?

Reply:
StringBuilder is not recommended anymore, it is okay to just concatenate

Reply:
not strictly true as far as I'm aware, anything with loops definitely should still be StringBuildered.
other than that it comes down to readability.

Up to you 👍

Reply:
Ah. every `+=` creates `StringBuilder` and `String` instances. My comment only makes sense for single line concatenation